velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp

```diff
@@ -317,5 +318,30 @@ TEST_F(ApproxDistinctTest, streaming) {
   ASSERT_EQ(result->asFlatVector<int64_t>()->valueAt(0), 1008);
 }
 
+// Ensure that we convert to dense HLL during merge when necessary.
+TEST_F(ApproxDistinctTest, memoryLeakInMerge) {
+  constexpr int kSize = 500;
+  auto nodeIdGen = std::make_shared<core::PlanNodeIdGenerator>();
+  std::vector<core::PlanNodePtr> sources;
+  for (int i = 0; i < 100; ++i) {
+    auto c0 =
+        makeFlatVector<int32_t>(kSize, [i](auto j) { return j + i * kSize; });
+    sources.push_back(PlanBuilder(nodeIdGen)
+                          .values({makeRowVector({c0})})
+                          .partialAggregation({}, {"approx_distinct(c0, 0.01)"})
+                          .planNode());
+  }
+  core::PlanNodeId finalAgg;
+  auto op = PlanBuilder(nodeIdGen)
+                .localMerge({}, std::move(sources))
+                .finalAggregation()
+                .capturePlanNodeId(finalAgg)
+                .planNode();
+  auto expected = makeFlatVector(std::vector<int64_t>({50311}));
+  auto task = assertQuery(op, {makeRowVector({expected})});
+  ASSERT_LT(
+      toPlanStats(task->taskStats()).at(finalAgg).peakMemoryBytes, 150'000);
```

Comment:
out of curiosity, where does this 150'000 magic number comes from?

Reply:
I just check the number before (500KB) & after (120KB) the fix and choose some number in between